src/main_nep/nep4.cu

```diff
@@ -144,12 +144,14 @@ static __global__ void find_descriptors_angular(
 
 NEP4::NEP4(char* input_dir, Parameters& para, int N, int N_times_max_NN_angular)
 {
+  printf("Creating NEP4\n");
   paramb.rc_angular = para.rc_angular;
   paramb.rcinv_angular = 1.0f / paramb.rc_angular;
   annmb.dim = para.dim;
   annmb.num_neurons1 = para.num_neurons1;
   paramb.num_types = para.num_types;
-  annmb.num_para = para.number_of_variables;
+  annmb.num_para = para.number_of_variables_ann;
```

Comment:
annmb.num_para = para.number_of_variables_ann;

Reply:
need to add para.number_of_variables_descriptor

Reply:
change line 153 to 
```
annmb.num_para = para.number_of_variables_ann + para.number_of_variables_descriptor;
```
This is because `annmb.num_para` include both the ANN parameters and the descriptor parameters (the `const float* c` in the  `ANN` structure.

Reply:
Yes, shouldn't it be like that?